### PR TITLE
[CDAP-17446] Replication Detail Page Redesign

### PR DESCRIPTION
**Pull request by elfenheart**
JIRAs: 
- https://cdap.atlassian.net/browse/CDAP-17460
- https://cdap.atlassian.net/browse/CDAP-17446

Build: https://builds.cask.co/browse/CDAP-UDUT896

### CODE REVIEW

File: cdap-ui/app/cdap/components/Replicator/Detail/ContentHeading/index.tsx [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+      },
+    },
+    contentHeading: {
+      display: 'grid',
+      gridTemplateColumns: '1fr 550px',
+      marginTop: '15px',
+      alignItems: 'center',
+    },
+    text: {
+      fontSize: '18px',
```
Comment (ajainarayanan) [inline cdap-ui/app/cdap/components/Replicator/Detail/ContentHeading/index.tsx]: nit: Are we introducing a different heading size?

File: cdap-ui/app/cdap/components/Replicator/Detail/DetailContent/index.tsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+const styles = (theme): StyleRules => {
+  return {
+    root: {
+      display: 'grid',
+      gridTemplateColumns: '300px 1fr',
+      alignItems: 'end',
+    },
+    link: {
+      color: theme.palette.grey[50],
+      fontSize: '25px',
```
Comment (ajainarayanan) [inline cdap-ui/app/cdap/components/Replicator/Detail/DetailContent/index.tsx]: Same here. We should use standardized headings. @leacd Is there any specific need to resize the headings here?

File: cdap-ui/app/cdap/components/Replicator/Detail/Monitoring/Metrics/parser.tsx [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  'user.dml.updates': 'updates',
+  'user.dml.deletes': 'deletes',
+  'user.dml.errors': 'errors',
+  'user.dml.latency.seconds': 'latency',
+  'user.dml.data.processed.bytes': 'dataReplicated',
+};
+
+const PRECISION = 2;
+
+const ONE_MIN_SECONDS = 60;
```
Comment (ajainarayanan) [inline cdap-ui/app/cdap/components/Replicator/Detail/Monitoring/Metrics/parser.tsx]: nit: This is already exported from helpers.

File: cdap-ui/app/cdap/components/Replicator/Detail/Monitoring/ThroughputLatencyGraphs/parser.ts [hunk 4]
```diff
@@ -44,10 +44,10 @@ interface IRawMetricData {
 }
 
 const METRIC_NAME_MAP = {
-  'user.dml.insert': 'inserts',
-  'user.dml.update': 'updates',
-  'user.dml.delete': 'deletes',
-  'user.dml.error': 'errors',
+  'user.dml.inserts': 'inserts',
```
Comment (ajainarayanan) [inline cdap-ui/app/cdap/components/Replicator/Detail/Monitoring/ThroughputLatencyGraphs/parser.ts:47]: nit: Seems like all the parsers have most of the metrics in common. Can we extract out the common metrics into one parser and reuse it. Would be useful if at some point in we change any of the metrics.

File: cdap-ui/app/cdap/components/Replicator/Detail/Overview/TablesList/index.tsx [hunk 5]
```diff
@@ -0,0 +1,330 @@
...
+                <TableCell textAlign="right">{tableMetrics.inserts}</TableCell>
+                <TableCell textAlign="right">{tableMetrics.updates}</TableCell>
+                <TableCell textAlign="right">{tableMetrics.deletes}</TableCell>
+              </TableRow>
+            );
+          })}
+        </TableBody>
+      </Table>
+
+      {/* <div className="grid-wrapper">
```
Comment (ajainarayanan) [inline cdap-ui/app/cdap/components/Replicator/Detail/Overview/TablesList/index.tsx]: nit: Will this be removed in a subsequent PR?

File: cdap-ui/app/cdap/components/Replicator/Detail/Overview/TablesList/index.tsx [hunk 6]
```diff
@@ -0,0 +1,330 @@
...
+                <InputAdornment position="start">
+                  <Search />
+                </InputAdornment>
+              ),
+            }}
+          />
+        </div>
+      </div>
+
+      <Table
```
Comment (ajainarayanan) [inline cdap-ui/app/cdap/components/Replicator/Detail/Overview/TablesList/index.tsx:227]: This is a nice start!

**Review by ajainarayanan — COMMENTED** [review]
Added some minor comment. Overall the changes look great! I think we are off to a good start on the table component!

File: cdap-ui/app/cdap/components/Replicator/Detail/Monitoring/ThroughputLatencyGraphs/parser.ts (hunk 4 above)
Comment (elfenheart) [reply cdap-ui/app/cdap/components/Replicator/Detail/Monitoring/ThroughputLatencyGraphs/parser.ts:47]: the parsers are different, the grouping of the metrics are different. There is some clean up and consolidation that can happen, will do that in subsequent PR

File: cdap-ui/app/cdap/components/Replicator/Detail/DetailContent/index.tsx (hunk 2 above)
Comment (elfenheart) [reply cdap-ui/app/cdap/components/Replicator/Detail/DetailContent/index.tsx]: the NavLink cannot apply the activeClass stylings if we use heading component here

File: cdap-ui/app/cdap/components/Replicator/Detail/Overview/TablesList/metricsParser.ts [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+
+const METRIC_MAP = {
+  'user.dml.inserts': 'inserts',
+  'user.dml.updates': 'updates',
+  'user.dml.deletes': 'deletes',
+  'user.dml.errors': 'errors',
+  'user.dml.latency.seconds': 'latency',
+};
+
+const ONE_MIN_SECONDS = 60;
```
Comment (ajainarayanan) [inline cdap-ui/app/cdap/components/Replicator/Detail/Overview/TablesList/metricsParser.ts]: nit: This needs to be changed as well.

File: cdap-ui/app/cdap/components/Replicator/Detail/DetailContent/index.tsx (hunk 2 above)
Comment (ajainarayanan) [reply cdap-ui/app/cdap/components/Replicator/Detail/DetailContent/index.tsx]: Wouldn't the Heading component be nested inside the NavLink? The activeClassStyles will still get applied to the H tags nested under the anchor tag right? Am I missing something.

**Review by ajainarayanan — APPROVED** [review]
Couple of questions. Otherwise 👍 LGTM

**Pull request merged**